### PR TITLE
REF: uses_mask in group_any_all

### PR DESCRIPTION
**Pull request by jbrockmendel**
any, all, and std go through GroupBy._get_cythonized_result instead of the more-standard WrappedCythonOp.  I'm trying to refactor any/all to use the other path, and as a step toward that am trying to make group_any_all follow the same patterns as the other functions in libgroupby.

The implementation here looks to me like it should behave identically to the existing implementation, but it fails a bunch of tests (mostly in tests/groupby/test_any_all.py).  Hoping @phofl can explain where I'm going wrong.

### CODE REVIEW

File: pandas/_libs/groupby.pyx [hunk 1]
```diff
@@ -627,12 +628,12 @@ def group_any_all(
...
                     continue
 
-                if nullable and mask[i, j]:
+                if uses_mask and mask[i, j]:
                     # Set the position as masked if `out[lab] != flag_val`, which
                     # would indicate True/False has not yet been seen for any/all,
                     # so by Kleene logic the result is currently unknown
                     if out[lab, j] != flag_val:
-                        out[lab, j] = -1
+                        result_mask[lab, j] = 1
```
Comment (phofl) [inline pandas/_libs/groupby.pyx:636]: For any: If NA is encountered as first value in the group you are setting the mask to 1 here but you don't reset it if you find another value in the group that is not NA. You'll have to update the mask if you find another value.

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -3733,14 +3731,12 @@ def blk_func(values: ArrayLike) -> ArrayLike:
...
+                    result_mask = result_mask[:, 0]
 
             if post_processing:
-                pp_kwargs: dict[str, bool | np.ndarray] = {}
-                pp_kwargs["nullable"] = isinstance(values, BaseMaskedArray)
-                if how == "std" and pp_kwargs["nullable"]:
-                    pp_kwargs["result_mask"] = result_mask
-
-                result = post_processing(result, inferences, **pp_kwargs)
+                result = post_processing(result, inferences, result_mask=result_mask)
```
Comment (phofl) [inline pandas/core/groupby/groupby.py:3739]: You'll probably have to remove the flag is_nullable for the std post_processing function (the default is False) and check for result_mask is not None

**Review by phofl — COMMENTED** [review]
Thanks for tackling this, this was on my todo list as well

File: pandas/_libs/groupby.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/groupby.pyx:636]: Oh OK.  I missed that we are checking `out[lab, j] != ...` here as opposed to `values[i, j] != ...`.  Thanks.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @jbrockmendel

**Pull request merged**